src/gpt_review/repositories/devops.py

```diff
@@ -0,0 +1,716 @@
+
+        Args:
+            diff (str): The patch of the PR.
+
+        Returns:
+            Dict[str, str]: The review.
+        """
+        link = os.getenv("LINK", link)
+        access_token = os.getenv("ADO_TOKEN", access_token)
+
+        if link and access_token:
+            review = _summarize_files(diff)
+
+            parsed_url = urlparse(link)
+
+            if "dev.azure.com" in parsed_url.netloc:
+                org = link.split("/")[3]
+                project = link.split("/")[4]
+                repo = link.split("/")[6]
+                pr_id = link.split("/")[8]
+            else:
+                org = link.split("/")[2].split(".")[0]
+                project = link.split("/")[3]
+                repo = link.split("/")[5]
+                pr_id = link.split("/")[7]
+
+            DevOpsClient(pat=access_token, org=org, project=project, repository_id=repo).update_pr(
+                pull_request_id=pr_id,
+                description=review,
+            )
+            return {"response": "PR posted"}
+
+        logging.warning("No PR to post too")
+        return {"response": "No PR to post too"}
+
+    @staticmethod
+    def get_pr_diff(patch_repo=None, patch_pr=None, access_token=None) -> str:
+        """
+        Get the diff of a PR.
+
+        Args:
+            patch_repo (str): The repo.
+            patch_pr (str): The PR.
+            access_token (str): The GitHub access token.
+
+        Returns:
+            str: The diff of the PR.
+        """
+
+
+class DevOpsFunction(DevOpsClient):
+    """Azure Function for process Service Messages from Azure DevOps."""
+
+    def handle(self, msg) -> None:
+        """
+        The main function for the Azure Function.
+
+        Args:
+            msg (func.QueueMessage): The Service Bus message.
+        """
+        body = msg.get_body().decode("utf-8")
+        logging.info("Python ServiceBus queue trigger processed message: %s", body)
+        if "copilot:summary" in body:
+            self._process_summary(body)
+        elif "copilot:" in body:
+            self._process_comment(body)
+
+    def _process_comment(self, body) -> None:
+        """
+        Process a comment from Copilot.
+
+        Args:
+            body (str): The Service Bus payload.
+        """
+        logging.info("Copilot Comment Alert Triggered")
+        payload = json.loads(body)
+
+        pr_id = self._get_pr_id(payload)
+
+        comment_id = self._get_comment_id(payload)
+
+        try:
+            diff = self.get_patch(pull_request_event=payload["resource"], pull_request_id=pr_id, comment_id=comment_id)
+        except Exception:
+            diff = self.get_patches(pull_request_event=payload["resource"])
+
+        logging.info("Copilot diff: %s", diff)
+        diff = "\n".join(diff)
+
+        question = f"""
+    {diff}
+
+    {_DevOpsClient.process_comment_payload(body)}
+    """
+
+        response = _ask(
+            question=question,
+            max_tokens=500,
```

Comment:
```suggestion
            max_tokens=1000,
```

parametrize or increase